core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/Run.java

```diff
@@ -48,241 +98,550 @@
  * @author Rob Davey
  * @since 0.0.2
  */
-@JsonSerialize(typing = JsonSerialize.Typing.STATIC)
-@JsonInclude(JsonInclude.Include.NON_NULL)
-@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.PROPERTY, property = "@class")
-@JsonIgnoreProperties({ "securityProfile", "submissionDocument" })
-public interface Run
```

Comment:
Yes, coding to interfaces lets you swap implementations, especially useful for testing. I'm not sure if there are many opportunities to create different implementations of model objects. The obvious case is creating a mock for testing, but modern mocking libraries let you mock concrete classes as well as interfaces.

I'm lukewarm on this. Other spring/hibernate apps I've written don't have interfaces for the model layer, and I didn't miss it. The model interfaces in MISO cause a mild hassle when they need to be updated, but in general the interfaces don't bother me.

I don't think our model layer needs the interfaces in general. And when there are specific cases, such as builders and multiple inheritance, we could add the interfaces for that specific purpose.